packages/mui-material/src/AppBar/AppBar.js

```diff
@@ -73,15 +73,17 @@ const AppBarRoot = styled(Paper, {
       position: 'relative',
     }),
     ...(ownerState.color === 'default' && {
-      backgroundColor: backgroundColorDefault,
-      color: theme.palette.getContrastText(backgroundColorDefault),
+      backgroundColor: theme.vars ? theme.vars.palette.AppBar.defaultBg : backgroundColorDefault,
+      color: theme.vars
+        ? theme.vars.palette.text.primary
+        : theme.palette.getContrastText(backgroundColorDefault),
     }),
     ...(ownerState.color &&
       ownerState.color !== 'default' &&
       ownerState.color !== 'inherit' &&
       ownerState.color !== 'transparent' && {
-        backgroundColor: theme.palette[ownerState.color].main,
-        color: theme.palette[ownerState.color].contrastText,
+        backgroundColor: (theme.vars || theme).palette[ownerState.color].main,
+        color: (theme.vars || theme).palette[ownerState.color].contrastText,
```

Comment:
Should we something with https://github.com/mui/material-ui/blob/989d1cae38ca7ace0c4abbd7477b5161232dbaf2/packages/mui-material/src/AppBar/AppBar.js#L91? Should we have some CSS variable that has the value `unset` for unsetting some CSS properties in cases like this?